src/Microsoft.AspNetCore.Routing.Abstractions/ILinkGenerator.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public interface ILinkGenerator
+    {
+        bool TryGetLink(LinkGeneratorContext context, out string link);
+
+        string GetLink(LinkGeneratorContext context);
+    }
```

Comment:
Future design note:

This is pretty bare-bones for now which is fine. We might want this to be an abstract class. It's very likely this surface area will grow.

Reply:
We also don't expect user code to implement this outside of maybe mocks for testing, so there's no real need to make it easy to do so.

Reply:
Filed: https://github.com/aspnet/Routing/issues/565